themes/powerline-multiline/powerline-multiline.base.bash

```diff
@@ -1,104 +1,107 @@
+			RIGHT_PROMPT+="$(set_color - "${LAST_SEGMENT_COLOR}") ${normal}"
+			((padding += 1))
+		fi
+		if [[ "${LAST_SEGMENT_COLOR}" -eq "${params[1]}" ]]; then
+			RIGHT_PROMPT+="$(set_color - "${LAST_SEGMENT_COLOR}")${POWERLINE_RIGHT_SEPARATOR_SOFT}${normal}"
+		else
+			RIGHT_PROMPT+="$(set_color "${params[1]}" "${LAST_SEGMENT_COLOR}")${POWERLINE_RIGHT_SEPARATOR}${normal}"
+		fi
+		((padding += 1))
+	fi
+
+	RIGHT_PROMPT+="$(set_color - "${params[1]}")${pad_before_segment}${params[0]}${normal}"
+
+	((padding += ${#pad_before_segment}))
+	((padding += ${#params[0]}))
+
+	((RIGHT_PROMPT_LENGTH += padding))
+	LAST_SEGMENT_COLOR="${params[1]}"
+	((SEGMENTS_AT_RIGHT += 1))
 }
 
 function __powerline_right_first_segment_padding {
-  RIGHT_PROMPT+="$(set_color - ${LAST_SEGMENT_COLOR}) ${normal}"
-  (( RIGHT_PROMPT_LENGTH += 1 ))
+	RIGHT_PROMPT+="$(set_color - "${LAST_SEGMENT_COLOR}") ${normal}"
+	((RIGHT_PROMPT_LENGTH += 1))
 }
 
 function __powerline_prompt_command {
-  local last_status="$?" ## always the first
-  local move_cursor_rightmost='\033[500C'
-
-  LEFT_PROMPT=""
-  RIGHT_PROMPT=""
-  RIGHT_PROMPT_LENGTH=${POWERLINE_PADDING}
-  SEGMENTS_AT_LEFT=0
-  SEGMENTS_AT_RIGHT=0
-  LAST_SEGMENT_COLOR=""
-
-  ## left prompt ##
-  for segment in $POWERLINE_LEFT_PROMPT; do
-    local info="$(__powerline_${segment}_prompt)"
-    [[ -n "${info}" ]] && __powerline_left_segment "${info}"
-  done
-
-  if [[ -n "${LEFT_PROMPT}" ]] && [[ "${POWERLINE_COMPACT_AFTER_LAST_SEGMENT}" -eq 0 ]]; then
-    __powerline_left_last_segment_padding
-  fi
-
-  [[ -n "${LEFT_PROMPT}" ]] && LEFT_PROMPT+="$(set_color ${LAST_SEGMENT_COLOR} -)${POWERLINE_LEFT_END}${normal}"
-
-  ## right prompt ##
-  if [[ -n "${POWERLINE_RIGHT_PROMPT}" ]]; then
-    # LEFT_PROMPT+="${move_cursor_rightmost}"
-    for segment in $POWERLINE_RIGHT_PROMPT; do
-      local info="$(__powerline_${segment}_prompt)"
-      [[ -n "${info}" ]] && __powerline_right_segment "${info}"
-    done
-
-    if [[ -n "${RIGHT_PROMPT}" ]] && [[ "${POWERLINE_COMPACT_BEFORE_FIRST_SEGMENT}" -eq 0 ]]; then
-      __powerline_right_first_segment_padding
-    fi
-
-    RIGHT_PAD=$(printf "%.s " $(seq 1 $RIGHT_PROMPT_LENGTH))
-    LEFT_PROMPT+="${RIGHT_PAD}${move_cursor_rightmost}"
-    LEFT_PROMPT+="\033[$(( ${#RIGHT_PAD} - 1 ))D"
-  fi
-
-  local prompt="${PROMPT_CHAR}"
-  if [[ "${POWERLINE_COMPACT_PROMPT}" -eq 0 ]]; then
-    prompt+=" "
-  fi
-
-  PS1="${LEFT_PROMPT}${RIGHT_PROMPT}\n$(__powerline_last_status_prompt ${last_status})${prompt}"
-
-  ## cleanup ##
-  unset LAST_SEGMENT_COLOR \
-        LEFT_PROMPT RIGHT_PROMPT RIGHT_PROMPT_LENGTH \
-        SEGMENTS_AT_LEFT SEGMENTS_AT_RIGHT
+	local last_status="$?" ## always the first
+	local move_cursor_rightmost='\033[500C'
+
+	LEFT_PROMPT=""
+	RIGHT_PROMPT=""
+	RIGHT_PROMPT_LENGTH=${POWERLINE_PADDING}
+	SEGMENTS_AT_LEFT=0
+	SEGMENTS_AT_RIGHT=0
+	LAST_SEGMENT_COLOR=""
+	_save-and-reload-history "${HISTORY_AUTOSAVE:-0}"
```

Comment:
/nitpick For consistency, let's add a blank line above this to separate it from the assignment block